lib/msnodesqlv8/request.js

```diff
@@ -234,16 +234,54 @@ class Request extends BaseRequest {
         return callback(null, [])
       }
 
-      let row = null
-      let columns = null
-      let recordset = null
       const recordsets = []
+      const errors = []
+      const errorHandlers = {}
       const output = {}
       const rowsAffected = []
+
+      let hasReturned = false
+      let row = null
+      let columns = null
+      let recordset = null
       let handleOutput = false
       let isChunkedRecordset = false
       let chunksBuffer = null
 
+      const handleError = (connection, info, moreErrors) => {
+        const doReturn = !moreErrors
+        if ((typeof info.sqlstate === 'string') && (info.sqlstate.toLowerCase() === '08s01')) {
+          connection.hasError = true
+        }
+
+        const err = new RequestError(info, 'EREQUEST')
+        err.code = 'EREQUEST'
+        err.state = info.sqlstate
+
+        if (this.stream) {
+          this.emit('error', err)
+        } else {
```

Comment:
Can combine doReturn && !hasReturned here into else if. Maybe have an else that throws as that's currently silently failing now.

Reply:
It's not silent failure, it's not meant to do anything if there are more errors expected

I can turn it into `else if` but for now I'm just copying the logic in `tedious.Request` and trying to get this green ;)